tests/test_mqttcm_connect.c

```diff
@@ -0,0 +1,589 @@
+/*----------------------------------------------------------------------------*/
+
+
+
+void test_fetchMqttParamsFromDB_success()
+{
+
+
+   fetchMqttParamsFromDB();
+   char locationId[256];
+    int locationIdResult = Get_Mqtt_LocationId(locationId);
+    CU_ASSERT_STRING_EQUAL(locationId, "12345678927e9a892c670333");
+    CU_ASSERT_EQUAL(locationIdResult, 0);
+
+    char* clientId = Get_Mqtt_ClientId();
+    CU_ASSERT_STRING_EQUAL(clientId, "123456789012");
+
+    char broker[256];
+    int brokerResult = Get_Mqtt_Broker(broker);
+    CU_ASSERT_STRING_EQUAL(broker, "localhost");
+    CU_ASSERT_EQUAL(brokerResult, 0);
+
+    char Port[32];
+    int portResult = Get_Mqtt_Port(Port);
+    CU_ASSERT_STRING_EQUAL(Port, "123");
+    CU_ASSERT_EQUAL(portResult, 0);
+}
+
+
+/*----------------------------------------------------------------------------*/
+/*                             Test case for getHostIPFromInterface           */
+/*----------------------------------------------------------------------------*/
+
+
+
+void test_getHostIPFromInterface_success()
+{
+	/*
+ 	char *interface = "global";
+    char *ip = NULL;
+    int result = getHostIPFromInterface(interface, &ip);
+
+    // Check if the function returns one indicating success
+    CU_ASSERT_EQUAL(result, 1);
+
+    // Check that the IP address is retrieved and not NULL
+    CU_ASSERT_PTR_NOT_NULL(ip);
+*/
+
+
+    struct ifaddrs *ifaddr, *ifa;
+    char *interface = NULL;
+    char *ip = NULL;
+    int result;
+
+    // Retrieve the list of network interfaces
+    if (getifaddrs(&ifaddr) == -1) {
+        CU_FAIL("Failed to get network interfaces");
+        return;
+    }
+
+    // Traverse the list of interfaces
+    for (ifa = ifaddr; ifa != NULL; ifa = ifa->ifa_next) {
+        // Check for valid interface and address
+        if (ifa->ifa_addr != NULL && ifa->ifa_addr->sa_family == AF_INET) {
+            // Extract the first valid interface
+            interface = strdup(ifa->ifa_name);
+            break;
+        }
+    }
+
+    // Free the memory allocated by getifaddrs
+    freeifaddrs(ifaddr);
+
+    // Check if a valid interface was found
+    if (interface == NULL) {
+        CU_FAIL("No valid network interface found");
+        return;
+    }
+
+    // Call the function under test
+    result = getHostIPFromInterface(interface, &ip);
+
+    // Check if the function returns one indicating success
+    CU_ASSERT_EQUAL(result, 1);
+
+    // Check that the IP address is retrieved and not NULL
+    CU_ASSERT_PTR_NOT_NULL(ip);
+	// Free the allocated memory
+    free(interface);
+    //free(ip);
+
+
+}
+
+void test_getHostIPFromInterface_failure()
+{
+    char *interface = "wlppp";
+    char *ip = NULL;
+    int result = getHostIPFromInterface(interface, &ip);
+
+    // Check if the function returns a zero value indicating failure
+    CU_ASSERT_EQUAL(result, 0);
+
+    // Check that the IP address is not retrieved (should be NULL)
+    CU_ASSERT_PTR_NULL(ip);
+
+
+}
+
+
```

Comment:
Normalize the space between different test case , some where it is 5 new lines between two successive lines and other place it is 2 or 1 newline